roles/openshift_cli/tasks/main.yml

```diff
@@ -8,11 +8,12 @@
     docker pull {{ openshift.common.cli_image }}:{{ openshift_image_tag }}
   when: openshift.common.is_containerized | bool
 
-- name: Create /usr/local/bin/openshift cli wrapper
-  template:
-    src: openshift.j2
-    dest: /usr/local/bin/openshift
-    mode: 0755
+- set_fact:
+    container_name: "openshift-cli-{{ 100000 | random }}"
+  when: openshift.common.is_containerized | bool
+
+- name: Copy openshift binary out of CLI image for use on the host
+  shell: docker create --name {{ container_name }} {{ openshift.common.cli_image }}:{{ openshift_image_tag }} && docker cp {{ container_name }}:/usr/bin/openshift /usr/local/bin/openshift && docker rm {{ container_name }}
```

Comment:
Hmm looking at how that worked it's possible oc was a different binary. I will update to copy each binary separately instead of symlinking, the old wrapper looks like it knew how to handle that better.

Is it the performance (time to copy binaries) you're concerned about?